File: examples/faker/hipster.test.js
import { describe, expect } from "https://jslib.k6.io/k6chaijs/4.3.4.3/index.js";
import { Faker } from "k6/x/faker";

export default function () {
  describe("hipster", (t) => {
    let faker = new Faker(11);
    expect(faker.hipster.word(), "word()").to.equal("offal");
    faker = new Faker(11);
    expect(faker.hipster.sentence(4), "sentence()").to.equal("Offal forage pinterest direct trade.");
    faker = new Faker(11);
    expect(faker.hipster.paragraph(1, 2, 4, "\n"), "paragraph()").to.equal(
      "Offal forage pinterest direct trade. Pug skateboard food truck flannel."
    );
  });
}

File: examples/faker/hipster_test.go
package faker

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func hipster(t *testing.T) goHipster {
	t.Helper()

	return &goHipsterImpl{fakeit: fakeit(t)}
}

func Test_goHipster(t *testing.T) {
	t.Parallel()
	must := noError(t)

	assert.Equal(t, "offal", must(hipster(t).wordMethod()))
	assert.Equal(t, "Offal forage pinterest direct trade.", must(hipster(t).sentenceMethod(4)))
	assert.Equal(
		t,
		"Offal forage pinterest direct trade. Pug skateboard food truck flannel.",
		must(hipster(t).paragraphMethod(1, 2, 4, "\n")),
	)
}
